src/DotNetWorker/Hosting/FunctionsApplication.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace Microsoft.Extensions.Hosting
+{
+    public class FunctionsApplication
```

Comment:
In AspNet... this implements `IHost` and others and hangs stuff off of it. Will we ever need to do that? Right now, I don't even use this other than for the static `CreateBuilder()`...

Reference: https://github.com/dotnet/aspnetcore/blob/main/src/DefaultBuilder/src/WebApplication.cs#L25

Reply:
I think it'll be a question of what extension methods we want to have hanging off of it. I know ASP.NET Core does a bunch of things only after the builder is built. We could probably get away without these for now and add them in if needed down the line.

Reply:
Yeah, ASP.NET Core implements a couple of different interfaces for the following:

- IApplicationBuilder to support registering ASP.NET Core middleware directly (`app.UseAuthorization`)
- IEndpointRouteBuilder to support calling the minimal API handlers directly (`app.MapGet`)
- `IHost` to support exposing the configuration of the underlying `HostApplicationBuilder` inner builder used in `WeApplicationBuilder`

You can probably get away with not implementing any of these although I imagine a possibly interesting feature where we can explore what it would look like for `FunctionsApplication` to implement `IApplicationBuilder` and support ASP.NET Core middleware more directly?

Reply:
@captainsafia -- Yeah we'd like to do that. We just haven't thought through the ramifications of it just yet (i.e. how can we prevent people from shooting themselves in the foot?).

But I do see us exposing the ASP.NET pipeline directly at some point in the near future.